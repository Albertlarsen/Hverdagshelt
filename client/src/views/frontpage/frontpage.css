.frontpage {
    height: 100%;
    background-image: url('https://i.imgur.com/5LhYoFW.jpg');
    background-size: cover;


}



.chooseCounty {
    padding-top: 20em;
}

