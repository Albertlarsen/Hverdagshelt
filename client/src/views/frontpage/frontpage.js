import { Col, Button, Form, FormGroup, Label, Grid } from 'react-bootstrap';
import { CountyService } from '../../services';
import { Component } from 'react';
import * as React from 'react';
import { Alert } from '../../widgets';
import ReactDOM from 'react-dom';
//import {County} from "../../classTypes";
import css from './frontpage.css';

import Select from 'react-select';
import Row from 'react-bootstrap/es/Row';

let countyService = new CountyService();
let imageUrl = '../../../public/resources/osloBackground.png';


export class Frontpage extends Component<Props, State> {

  constructor(props) {
    super(props);
    this.state = {
      isLoaded: false,
      choosen: { label: 'Bergen', countyId: 1 },
      values: [
        { label: 'Bergen', countyId: 1 }
      ]
    };

    this.handleChangeCounty = this.handleChangeCounty.bind(this);
  }


  handleChangeCounty(e: Object) {
    console.log(e.value);
    this.setState({
      choosen: JSON.parse(e.value)
    });


  };

  componentWillMount() {
    var arr = [];
    countyService
      .getCounties()
      .then(county2 => {
        county2.map(e => {
          var elem = {
            name: e.name,
            countyId: e.countyId
          };
          arr = arr.concat(elem);

        });
        this.setState({
          values: arr
        });
      })
      .catch((error: Error) => Alert.danger(error.message));

  }

  render() {
    let optionTemplate = this.state.values.map(v => {
      var data = { label: v.name, value: v.countyId, countyId: v.countyId };
      return (data);
    });
    return (
      <div className="frontpage">
        <Grid>
          <FormGroup>
            <Select
              placeholder={'Klikk her for å velge kommune'}
              options={optionTemplate}
              className="frontpage-county"
              onChange={this.handleChangeCounty}
            />
          </FormGroup>
          <Button className={'frontpage-button'} href={'/#home/' + this.state.choosen}>Gå videre!</Button>
        </Grid>
      </div>
    );
  }

}
