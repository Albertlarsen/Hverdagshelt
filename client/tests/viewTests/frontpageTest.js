// @flow

import {shallow} from "enzyme/build";
import * as React from "react";
import {Frontpage} from "../../src/views/frontpage/frontpage";

describe('Test for Frontpage view', () => {
    const wrapper = shallow(<Frontpage />);

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });
});