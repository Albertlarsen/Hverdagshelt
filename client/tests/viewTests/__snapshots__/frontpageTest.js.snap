// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Frontpage view renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Frontpage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Grid
        bsClass="container"
        className="chooseCounty"
        componentClass="div"
        fluid={false}
      >
        <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={1}
          />
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={10}
          >
            <FormGroup
              bsClass="form-group"
            >
              <StateManager
                className="frontpage-county"
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "countyId": 1,
                      "label": undefined,
                      "value": 1,
                    },
                  ]
                }
                placeholder="Klikk her for å velge kommune"
              />
            </FormGroup>
            <div
              align="center"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                className="frontpage-button"
                disabled={false}
                onClick={[Function]}
              >
                Gå videre!
              </Button>
            </div>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={1}
          />
        </Form>
      </Grid>,
      "className": "frontpage",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bsClass": "container",
        "children": <Form
          bsClass="form"
          componentClass="form"
          horizontal={false}
          inline={false}
        >
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={1}
          />
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={10}
          >
            <FormGroup
              bsClass="form-group"
            >
              <StateManager
                className="frontpage-county"
                defaultInputValue=""
                defaultMenuIsOpen={false}
                defaultValue={null}
                onChange={[Function]}
                options={
                  Array [
                    Object {
                      "countyId": 1,
                      "label": undefined,
                      "value": 1,
                    },
                  ]
                }
                placeholder="Klikk her for å velge kommune"
              />
            </FormGroup>
            <div
              align="center"
            >
              <Button
                active={false}
                block={false}
                bsClass="btn"
                bsStyle="primary"
                className="frontpage-button"
                disabled={false}
                onClick={[Function]}
              >
                Gå videre!
              </Button>
            </div>
          </Col>
          <Col
            bsClass="col"
            componentClass="div"
            md={4}
            xs={1}
          />
        </Form>,
        "className": "chooseCounty",
        "componentClass": "div",
        "fluid": false,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "form",
          "children": Array [
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />,
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={10}
            >
              <FormGroup
                bsClass="form-group"
              >
                <StateManager
                  className="frontpage-county"
                  defaultInputValue=""
                  defaultMenuIsOpen={false}
                  defaultValue={null}
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "countyId": 1,
                        "label": undefined,
                        "value": 1,
                      },
                    ]
                  }
                  placeholder="Klikk her for å velge kommune"
                />
              </FormGroup>
              <div
                align="center"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  className="frontpage-button"
                  disabled={false}
                  onClick={[Function]}
                >
                  Gå videre!
                </Button>
              </div>
            </Col>,
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />,
          ],
          "componentClass": "form",
          "horizontal": false,
          "inline": false,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "componentClass": "div",
              "md": 4,
              "xs": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "children": Array [
                <FormGroup
                  bsClass="form-group"
                >
                  <StateManager
                    className="frontpage-county"
                    defaultInputValue=""
                    defaultMenuIsOpen={false}
                    defaultValue={null}
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "countyId": 1,
                          "label": undefined,
                          "value": 1,
                        },
                      ]
                    }
                    placeholder="Klikk her for å velge kommune"
                  />
                </FormGroup>,
                <div
                  align="center"
                >
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    className="frontpage-button"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Gå videre!
                  </Button>
                </div>,
              ],
              "componentClass": "div",
              "md": 4,
              "xs": 10,
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "bsClass": "form-group",
                  "children": <StateManager
                    className="frontpage-county"
                    defaultInputValue=""
                    defaultMenuIsOpen={false}
                    defaultValue={null}
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "countyId": 1,
                          "label": undefined,
                          "value": 1,
                        },
                      ]
                    }
                    placeholder="Klikk her for å velge kommune"
                  />,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "className": "frontpage-county",
                    "defaultInputValue": "",
                    "defaultMenuIsOpen": false,
                    "defaultValue": null,
                    "onChange": [Function],
                    "options": Array [
                      Object {
                        "countyId": 1,
                        "label": undefined,
                        "value": 1,
                      },
                    ],
                    "placeholder": "Klikk her for å velge kommune",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "align": "center",
                  "children": <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    className="frontpage-button"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Gå videre!
                  </Button>,
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "active": false,
                    "block": false,
                    "bsClass": "btn",
                    "bsStyle": "primary",
                    "children": "Gå videre!",
                    "className": "frontpage-button",
                    "disabled": false,
                    "onClick": [Function],
                  },
                  "ref": null,
                  "rendered": "Gå videre!",
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "bsClass": "col",
              "componentClass": "div",
              "md": 4,
              "xs": 1,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Grid
          bsClass="container"
          className="chooseCounty"
          componentClass="div"
          fluid={false}
        >
          <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={10}
            >
              <FormGroup
                bsClass="form-group"
              >
                <StateManager
                  className="frontpage-county"
                  defaultInputValue=""
                  defaultMenuIsOpen={false}
                  defaultValue={null}
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "countyId": 1,
                        "label": undefined,
                        "value": 1,
                      },
                    ]
                  }
                  placeholder="Klikk her for å velge kommune"
                />
              </FormGroup>
              <div
                align="center"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  className="frontpage-button"
                  disabled={false}
                  onClick={[Function]}
                >
                  Gå videre!
                </Button>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />
          </Form>
        </Grid>,
        "className": "frontpage",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "bsClass": "container",
          "children": <Form
            bsClass="form"
            componentClass="form"
            horizontal={false}
            inline={false}
          >
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={10}
            >
              <FormGroup
                bsClass="form-group"
              >
                <StateManager
                  className="frontpage-county"
                  defaultInputValue=""
                  defaultMenuIsOpen={false}
                  defaultValue={null}
                  onChange={[Function]}
                  options={
                    Array [
                      Object {
                        "countyId": 1,
                        "label": undefined,
                        "value": 1,
                      },
                    ]
                  }
                  placeholder="Klikk her for å velge kommune"
                />
              </FormGroup>
              <div
                align="center"
              >
                <Button
                  active={false}
                  block={false}
                  bsClass="btn"
                  bsStyle="primary"
                  className="frontpage-button"
                  disabled={false}
                  onClick={[Function]}
                >
                  Gå videre!
                </Button>
              </div>
            </Col>
            <Col
              bsClass="col"
              componentClass="div"
              md={4}
              xs={1}
            />
          </Form>,
          "className": "chooseCounty",
          "componentClass": "div",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "bsClass": "form",
            "children": Array [
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={1}
              />,
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={10}
              >
                <FormGroup
                  bsClass="form-group"
                >
                  <StateManager
                    className="frontpage-county"
                    defaultInputValue=""
                    defaultMenuIsOpen={false}
                    defaultValue={null}
                    onChange={[Function]}
                    options={
                      Array [
                        Object {
                          "countyId": 1,
                          "label": undefined,
                          "value": 1,
                        },
                      ]
                    }
                    placeholder="Klikk her for å velge kommune"
                  />
                </FormGroup>
                <div
                  align="center"
                >
                  <Button
                    active={false}
                    block={false}
                    bsClass="btn"
                    bsStyle="primary"
                    className="frontpage-button"
                    disabled={false}
                    onClick={[Function]}
                  >
                    Gå videre!
                  </Button>
                </div>
              </Col>,
              <Col
                bsClass="col"
                componentClass="div"
                md={4}
                xs={1}
              />,
            ],
            "componentClass": "form",
            "horizontal": false,
            "inline": false,
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "componentClass": "div",
                "md": 4,
                "xs": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "children": Array [
                  <FormGroup
                    bsClass="form-group"
                  >
                    <StateManager
                      className="frontpage-county"
                      defaultInputValue=""
                      defaultMenuIsOpen={false}
                      defaultValue={null}
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "countyId": 1,
                            "label": undefined,
                            "value": 1,
                          },
                        ]
                      }
                      placeholder="Klikk her for å velge kommune"
                    />
                  </FormGroup>,
                  <div
                    align="center"
                  >
                    <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="primary"
                      className="frontpage-button"
                      disabled={false}
                      onClick={[Function]}
                    >
                      Gå videre!
                    </Button>
                  </div>,
                ],
                "componentClass": "div",
                "md": 4,
                "xs": 10,
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "bsClass": "form-group",
                    "children": <StateManager
                      className="frontpage-county"
                      defaultInputValue=""
                      defaultMenuIsOpen={false}
                      defaultValue={null}
                      onChange={[Function]}
                      options={
                        Array [
                          Object {
                            "countyId": 1,
                            "label": undefined,
                            "value": 1,
                          },
                        ]
                      }
                      placeholder="Klikk her for å velge kommune"
                    />,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "className": "frontpage-county",
                      "defaultInputValue": "",
                      "defaultMenuIsOpen": false,
                      "defaultValue": null,
                      "onChange": [Function],
                      "options": Array [
                        Object {
                          "countyId": 1,
                          "label": undefined,
                          "value": 1,
                        },
                      ],
                      "placeholder": "Klikk her for å velge kommune",
                    },
                    "ref": null,
                    "rendered": null,
                    "type": [Function],
                  },
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "host",
                  "props": Object {
                    "align": "center",
                    "children": <Button
                      active={false}
                      block={false}
                      bsClass="btn"
                      bsStyle="primary"
                      className="frontpage-button"
                      disabled={false}
                      onClick={[Function]}
                    >
                      Gå videre!
                    </Button>,
                  },
                  "ref": null,
                  "rendered": Object {
                    "instance": null,
                    "key": undefined,
                    "nodeType": "class",
                    "props": Object {
                      "active": false,
                      "block": false,
                      "bsClass": "btn",
                      "bsStyle": "primary",
                      "children": "Gå videre!",
                      "className": "frontpage-button",
                      "disabled": false,
                      "onClick": [Function],
                    },
                    "ref": null,
                    "rendered": "Gå videre!",
                    "type": [Function],
                  },
                  "type": "div",
                },
              ],
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "bsClass": "col",
                "componentClass": "div",
                "md": 4,
                "xs": 1,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
